flytekit/clis/helpers.py

```diff
@@ -89,9 +89,14 @@ def str2bool(str):
 def _hydrate_identifier(
     project: str, domain: str, version: str, identifier: _identifier_pb2.Identifier
 ) -> _identifier_pb2.Identifier:
-    identifier.project = identifier.project or project
-    identifier.domain = identifier.domain or domain
-    identifier.version = identifier.version or version
+    if not identifier.project or identifier.project == "{{ .project }}":
```

Comment:
should these not be regexes?

Reply:
if you want example regex look at https://github.com/lyft/flytekit/blob/annotations/flytekit/annotated/base_sql_task.py#L15

Reply:
perhaps, but in the case of the project field the only templatized variable that can ever be used is project

Reply:
ya but what about whitespaces?

Reply:
also this should be changed right?

Reply:
derp, yes. had my test case backwards 🤦

Reply:
re: whitespace, the templating only ever comes from flytesdk right? presumably we can trust that to properly assign the constants, whitespace and all?